source/native-plugins/midi-transpose.c

```diff
@@ -50,57 +52,85 @@ static void miditranspose_cleanup(NativePluginHandle handle)
 
 static uint32_t miditranspose_get_parameter_count(NativePluginHandle handle)
 {
-    return 1;
+    return 2;
 
     // unused
     (void)handle;
 }
 
 static const NativeParameter* miditranspose_get_parameter_info(NativePluginHandle handle, uint32_t index)
 {
-    if (index != 0)
+    if (index >= miditranspose_get_parameter_count(handle))
         return NULL;
 
-    static NativeParameter param;
-
-    param.name  = "Octaves";
-    param.unit  = NULL;
-    param.hints = NATIVE_PARAMETER_IS_ENABLED|NATIVE_PARAMETER_IS_AUTOMABLE|NATIVE_PARAMETER_IS_INTEGER;
-    param.ranges.def = 0.0f;
-    param.ranges.min = -8.0f;
-    param.ranges.max = 8.0f;
-    param.ranges.step = 1.0f;
-    param.ranges.stepSmall = 1.0f;
-    param.ranges.stepLarge = 1.0f;
-    param.scalePointCount = 0;
-    param.scalePoints     = NULL;
-
-    return &param;
-
-    // unused
-    (void)handle;
+    static NativeParameter param[] =
```

Comment:
I wasn't sure if I should use a switch-case here, so I made 'param' an array.  Happy to know if this is good or bad.

Reply:
better to just see what the other plugins do and just copy them.
no shame in copy&pasting :)

Reply:
Haha yes!  For some reason this only occurred to me after the fact, but I left it as-is, because it compiled and ran as intended, and on the off-chance that this might be a better way, I wanted to show you :-)